### DISCUSSION TITLE
Function calling with assistants api sample?

### DISCUSSION THREAD

**Discussion by lofti198** [higty/higlabo#92] [Q&A] [answered]
Hello! Once more thank you for your great package! Is there somewhere find and example of function calling within assistants usage scenario ? From what I found, there are just chat completions examples. Thank you in advance

**Comment by higty** [comment, marked as answer]
I update HigLabo.OpenAI and added sample code.
https://github.com/higty/higlabo/blob/master/Net8/HigLabo.OpenAI.SampleConsoleApp/OpenAIPlayground.cs#L444

**Reply by lofti198** [reply]
Its awesome! Thank you for such fast adding this feature as well as for great example!

**Reply by higty** [reply]
@lofti198 
You can also call OpenAI GPT-4o model from today!

**Reply by lofti198** [reply]
Great! Just wanted to ask about this. By the way - I am currently challenged to implement the followin task, maybe you can suggest some sample from your library. I want to send several messages one by one and finally ask to make one general text from them (this is my approach to go around request token limitation for big models). So the final dialog should look like this (tried it in GPT web version and also here https://platform.openai.com/playground/chat?models=gpt-4o got the same result):

```
i will send you several messages and then I will ask you to create one general text from them
ChatGPT
Sure, go ahead and send the messages. I'll be ready to create a general text from them once you've sent everything.

User
Once upon a time, lived a frog
ChatGPT
Got it. Please continue with the next messages.

User
She had 5 children
ChatGPT
Okay, noted. Please send the next message when ready.

User
She was very happy

ChatGPT
Understood. Please send the next message.

User
Now join them all in one text
ChatGPT
Once upon a time, there lived a frog. She had 5 children and was very happy.
```

**Comment by higty** [comment]
User
Now join them all in one text
ChatGPT
Once upon a time, there lived a frog. She had 5 children and was very happy.

You change above to below

User
Please join below json data

{
    "Message1": "Once upon a time, lived a frog",
    "Message2": "She had 5 children",
    "Message3": "She was very happy",
    "Message4": "....",
}

Is it sute your requirement?
I hope that It may be good result!

**Reply by lofti198** [reply]
Hi! Thank you for your example! However will it finally send messages one by one (like different requests) to openai? And aslo consider current conversation ID? So that API understand, that those are messages from the one conversation.  My task is to input several texts and get their overall summary. When I send more than 3k tokens in one message, it gives me error (text too long), even with a model context of 128k (chatgpt4o).  So I want to send the 1st message like this:

```

I will send you text chunks in json messages: { command:"take_next_chunk", chunk:"..." } 

After message with next chunk, just save it to your memory and require next chunk in a short phrase. 

 After all chunks I will send you the following json message: { command:"process" }  

After the "process" method - give me the overall summary of all the chunks, I have sent before.`

```


then send a number of such messages:
`{ command:"take_next_chunk", chunk:"..." } `


and finally this one
`{ command:"process" }  `

But for each message I need to put the same conversation ID.

**Comment by lofti198** [comment]
Great thank you!
